common/OpTestUtil.py

```diff
@@ -2353,6 +2353,123 @@ def prepare_source(self,spec_file, host, dest_path, package, build_option=None,
+            connection.host_run_command("git bisect good {} ".format(good_tag))
+        else:
+            connection.host_run_command("git bisect good {} ".format(good_commit))
+        connection.host_run_command(" git bisect bad ")
+        connection.host_run_command("chmod +x ./make.sh ")
+        commit =  connection.host_run_command(" git bisect run ./make.sh")
+        badCommit = [word for word in commit if word.endswith("is the first bad commit")]
+        badCommit= badCommit[0].split()[0]
+        email = self.get_email(badCommit)
+        connection.host_run_command("git bisect log")
+        connection.host_run_command("git bisect reset")
+        return email, badCommit
+
+    def get_commit_message(self, linux_path, commit_sha):
+        connection = self.conf.host()
+        try:
+            connection.host_run_command(" if [ '$(pwd)' != {} ]; then cd {} || exit 1 ; fi ".format(linux_path,linux_path))
+            commit_message = connection.host_run_command("git log -n 1 --pretty=format:%s {} | sed -r 's/\x1B\[[0-9:]*[JKsu]//g'".format(commit_sha))
+        except subprocess.CalledProcessError:
+            commit_message = None
+        return commit_message[0].strip()
+
+    def format_email(self, linux_path , repo):
+        connection = self.conf.host()
+        machine_type = connection.host_run_command("uname -m")
+        gcc_version = connection.host_run_command("gcc --version")[0]
+        kernel_version = connection.host_run_command("uname -r")
+        try:
+            with open("output.json", "r") as file:
+                data = json.load(file)
+                error_message = data.get("error", "")
+                err_long = data.get("err_msg","")
+                commit = str(data.get("commit", ""))[:7]
+        except FileNotFoundError:
+            print("Error: output.json not found.")
+            error_message = ""
+            commit = ""
+        fix_description = self.get_commit_message(linux_path,commit)
+        if "netdev/net" in repo:
+            linux = "netdev/net"
+        elif "netdev/net-next" in repo:
+            linux = "netdev/net-next"
+        elif "mkp/scsi" in repo :
+            linux = "scsi/scsi-queue"
+        elif "torvalds/linux" in repo:
+            linux = "mainline/master"
+        elif "next/linux-next" in repo:
+            linux = "linux-next/master"
+        else:
+            linux = "linux"
+        subject = "[{}][bisected {}] [{}] build fail with error: {}".format(linux,commit, machine_type, error_message)
+        body = """
+        Greetings,
+
+        Today's {} kernel fails to build on {} machine.
+
+        Kernel build fail at error: {}
+        {}
+
+
+        Kernel Version: {}
+        Machine Type: {}
+        gcc: {}
+        Bisected Commit: {}
+
+        kernel builds fine when the bad commit ({})  is reverted
+        {} - {}
+
+        --
+        Regards
+        Linux CI
```

Comment:
Please remove unnecessary whitespace  as mail content having more whitespace not loks good

Reply:
removed extra lines